File: 2423-remove-letter-to-equalize-frequency/2423-remove-letter-to-equalize-frequency.cpp
class Solution {
public:
    bool equalFrequency(string word) {
        int freq[26] = {0};
        int maxFreq = 0;
        for(char c : word){
            freq[c - 'a']++;
            maxFreq = max(maxFreq, freq[c - 'a']);
        }
        int maxCount = 0;
        int maxMinusOneCount = 0;

        for(int i = 0; i < 26; i++) {
            if(freq[i] == 0) continue;
            if(freq[i] == maxFreq) maxCount++;
            if((freq[i] == maxFreq - 1)) maxMinusOneCount++;
            if(freq[i] < maxFreq - 1) return false;

        }
        return maxMinusOneCount == 1 || maxCount == 1 || (maxMinusOneCount == 0 && maxFreq == 1);
    
    }


};

File: 2423-remove-letter-to-equalize-frequency/2423-remove-letter-to-equalize-frequency.java
class Solution {
    public boolean equalFrequency(String word) {
        int[] f = new int[26];
        int maxFreq = 0;
        for(char c : word.toCharArray()){
            f[c - 'a']++;
            maxFreq = Math.max(maxFreq, f[c - 'a']);
        }
        int maxCount = 0;
        int maxMinusOneCount = 0;

        for(int i = 0; i < 26; i++) {
            if(f[i] == 0) continue;
            if(f[i] == maxFreq) maxCount++;
            if((f[i] == maxFreq - 1)) maxMinusOneCount++;
            if(f[i] < maxFreq - 1) return false;

        }
        return maxMinusOneCount == 1 || maxCount == 1 || (maxMinusOneCount == 0 && maxFreq == 1);
    }
}
